File: surface/widgets/swiftui/components/widgets/navigation/Accordion.swift
// ============================================================
// Clef Surface SwiftUI Widget — Accordion
//
// Vertically stacked collapsible sections. Each section has a
// trigger heading and expandable content panel with animation.
// Supports single or multiple expanded sections.
// ============================================================

import SwiftUI

// --------------- Types ---------------

struct AccordionItem: Identifiable {
    let id: String
    let title: String
    let content: String
}

// --------------- Component ---------------

/// Accordion view with vertically stacked collapsible sections.
///
/// - Parameters:
///   - items: Array of collapsible sections.
///   - multiple: Allow multiple sections open simultaneously.
///   - defaultOpen: IDs of initially expanded sections.
///   - onChange: Callback when expanded sections change.
struct AccordionView: View {
    var items: [AccordionItem]
    var multiple: Bool = false
    var defaultOpen: [String] = []
    var onChange: (([String]) -> Void)? = nil

    @State private var openIds: Set<String> = []

    var body: some View {
        VStack(spacing: 0) {
            ForEach(Array(items.enumerated()), id: \.element.id) { index, item in
                let isOpen = openIds.contains(item.id)

                // Trigger row
                SwiftUI.Button(action: {
                    withAnimation(.easeInOut(duration: 0.2)) {
                        if isOpen {
                            openIds.remove(item.id)
                        } else {
                            if multiple {
                                openIds.insert(item.id)
                            } else {
                                openIds = [item.id]
                            }
                        }
                        onChange?(Array(openIds))
                    }
                }) {
                    HStack {
                        Text(item.title)
                            .font(.subheadline)
                            .fontWeight(.semibold)
                            .foregroundColor(.primary)
                        Spacer()
                        Image(systemName: "chevron.down")
                            .rotationEffect(.degrees(isOpen ? 180 : 0))
                            .foregroundColor(.secondary)
                    }
                    .padding(.vertical, 12)
                    .padding(.horizontal, 16)
                }
                .buttonStyle(.plain)
                .accessibilityLabel(item.title)
                .accessibilityHint(isOpen ? "Collapse" : "Expand")

                // Expandable content
                if isOpen {
                    Text(item.content)
                        .font(.body)
                        .padding(.horizontal, 16)
                        .padding(.bottom, 12)
                        .transition(.opacity)
                }

                if index < items.count - 1 {
                    Divider()
                }
            }
        }
        .onAppear {
            openIds = Set(defaultOpen)
        }
    }
}

File: surface/widgets/nativescript/components/widgets/navigation/Accordion.ts
// ============================================================
// Clef Surface NativeScript Widget — Accordion
//
// Vertically stacked collapsible sections with expand/collapse.
// ============================================================

import { StackLayout, Label, Button } from '@nativescript/core';

export interface AccordionItem {
  value: string;
  trigger: string;
  content: string;
  disabled?: boolean;
}

export interface AccordionProps {
  items: AccordionItem[];
  value?: string[];
  defaultValue?: string[];
  multiple?: boolean;
  collapsible?: boolean;
  disabled?: boolean;
  onValueChange?: (value: string[]) => void;
  variant?: string;
  size?: string;
}

export function createAccordion(props: AccordionProps): StackLayout {
  const {
    items, value, defaultValue = [], multiple = false,
    collapsible = true, disabled = false, onValueChange,
    variant, size,
  } = props;

  let expandedItems = value ?? [...defaultValue];
  const container = new StackLayout();
  container.className = 'clef-widget-accordion';

  for (const item of items) {
    const section = new StackLayout();
    section.className = 'clef-accordion-item';
    const isExpanded = expandedItems.includes(item.value);
    const isDisabled = disabled || !!item.disabled;

    const trigger = new Button();
    trigger.text = `${isExpanded ? '\u25BC' : '\u25B6'} ${item.trigger}`;
    trigger.isEnabled = !isDisabled;
    trigger.accessibilityRole = 'button';
    trigger.accessibilityState = { expanded: isExpanded };

    const content = new StackLayout();
    content.visibility = isExpanded ? 'visible' : 'collapsed';
    content.padding = '8 16';
    const contentLabel = new Label();
    contentLabel.text = item.content;
    contentLabel.textWrap = true;
    content.addChild(contentLabel);

    trigger.on('tap', () => {
      if (isDisabled) return;
      const wasExpanded = expandedItems.includes(item.value);
      if (wasExpanded) {
        if (!collapsible && expandedItems.length === 1) return;
        expandedItems = expandedItems.filter((v: string) => v !== item.value);
      } else {
        expandedItems = multiple ? [...expandedItems, item.value] : [item.value];
      }
      onValueChange?.(expandedItems);
    });

    section.addChild(trigger);
    section.addChild(content);
    container.addChild(section);
  }
  return container;
}

export default createAccordion;
